### PR TITLE
Migrate macrobenchmarks to null safety

### PR DESCRIPTION
**Pull request by Abhishek01039**
Related to https://github.com/flutter/flutter/issues/84014

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dev/benchmarks/macrobenchmarks/lib/src/cull_opacity.dart
Comment (jmagman) [inline dev/benchmarks/macrobenchmarks/lib/src/cull_opacity.dart:19]: Can you move `addListener` back into `initState()`?

File: dev/benchmarks/macrobenchmarks/lib/src/filtered_child_animation.dart
Comment (jmagman) [inline dev/benchmarks/macrobenchmarks/lib/src/filtered_child_animation.dart:34]: `late FilterType?`?

File: dev/benchmarks/macrobenchmarks/lib/src/filtered_child_animation.dart
Comment (jmagman) [inline dev/benchmarks/macrobenchmarks/lib/src/filtered_child_animation.dart:36]: Maybe initialize these to false, but keep the widget initialization in `initState`

File: dev/benchmarks/macrobenchmarks/lib/src/filtered_child_animation.dart
Comment (jmagman) [inline dev/benchmarks/macrobenchmarks/lib/src/filtered_child_animation.dart:194]:
```suggestion
                  onChanged: (bool? b) => setState(() => _complexChild = b == true),
```

File: dev/benchmarks/macrobenchmarks/lib/src/post_backdrop_filter.dart
Comment (jmagman) [inline dev/benchmarks/macrobenchmarks/lib/src/post_backdrop_filter.dart:85]: Audit these

File: dev/benchmarks/macrobenchmarks/lib/src/web/bench_build_image.dart
Comment (jmagman) [inline dev/benchmarks/macrobenchmarks/lib/src/web/bench_build_image.dart:67]: For all of these: the initialization can be set in the `late final`, but keep the methods that act on it in `initState`, so move the `repeat()` part back into the `initState`

**Review by jmagman — CHANGES_REQUESTED** [review]

File: dev/benchmarks/macrobenchmarks/lib/src/filtered_child_animation.dart
Comment (Abhishek01039) [reply dev/benchmarks/macrobenchmarks/lib/src/filtered_child_animation.dart:34]: Because 
```
void _setFilterType(FilterType type, bool selected) {
    setState(() => _filterType = selected ? type : null);
  }
```

This function can assign FilterType to null

File: dev/benchmarks/macrobenchmarks/lib/src/web/bench_build_image.dart
Comment (Abhishek01039) [reply dev/benchmarks/macrobenchmarks/lib/src/web/bench_build_image.dart:67]: Sure thanks

File: dev/benchmarks/complex_layout/ios/Flutter/Debug.xcconfig
Comment (Piinks) [inline dev/benchmarks/complex_layout/ios/Flutter/Debug.xcconfig:1]: Is this change necessary?

File: dev/benchmarks/complex_layout/ios/Flutter/Release.xcconfig
Comment (Piinks) [inline dev/benchmarks/complex_layout/ios/Flutter/Release.xcconfig:1]: Same here

File: dev/benchmarks/macrobenchmarks/lib/src/filtered_child_animation.dart
Comment (jmagman) [reply dev/benchmarks/macrobenchmarks/lib/src/filtered_child_animation.dart:34]: I see, you want it lazily instantiated, but it's nullable.

File: dev/benchmarks/macrobenchmarks/lib/src/cull_opacity.dart
Comment (jmagman) [inline dev/benchmarks/macrobenchmarks/lib/src/cull_opacity.dart:26]: whitespace
```suggestion
      setState(() {});
```

**Review by jmagman — APPROVED** [review]
LGTM once @Piinks is happy.

File: dev/benchmarks/macrobenchmarks/lib/src/filtered_child_animation.dart
Comment (Abhishek01039) [reply dev/benchmarks/macrobenchmarks/lib/src/filtered_child_animation.dart:34]: yes exactly.

File: dev/benchmarks/complex_layout/ios/Flutter/Debug.xcconfig
Comment (Abhishek01039) [reply dev/benchmarks/complex_layout/ios/Flutter/Debug.xcconfig:1]: Nope this changes is made by vscode bydefault

Comment (0503256405) [reply dev/benchmarks/complex_layout/ios/Flutter/Debug.xcconfig]: 2000umairkhan 5673568467

**Review by 0503256405 — APPROVED** [review]

**Review by 0503256405 — APPROVED** [review]
2000umairkhan 5673568467

File: dev/benchmarks/macrobenchmarks/lib/src/filtered_child_animation.dart
Comment (0503256405) [reply dev/benchmarks/macrobenchmarks/lib/src/filtered_child_animation.dart:34]: 2000umairkhan 5673568467 0503 2564 05

**Review by Piinks — APPROVED** [review]
Flutter_LGTM

**Comment by Abhishek01039** [comment]
Hello @Piinks 
`Linux web_benchmarks_html` is stuck here now.

**Comment by jmagman** [comment]
The web_benchmarks_html tests are timing out in macrobenchmarks.

@Abhishek01039 Can you try running these tests locally to see if they pass on your machine?
```
$ dart bin/run.dart -t web_benchmarks_html     
```

**Comment by Abhishek01039** [comment]
I have tried @jmagman 
it's failing in my local machine as well.

**Comment by jmagman** [comment]
> I have tried @jmagman
> it's failing in my local machine as well.

Then it needs to be fixed before this change can merge 🙂

**Comment by Abhishek01039** [comment]
Can you please guide me on this issue ?
thanks

**Comment by Abhishek01039** [comment]
```[CHROME STDERR]:Fontconfig warning: "/etc/fonts/fonts.conf", line 100: unknown element "blank"```

This seems an error @jmagman

**Comment by jmagman** [comment]
Same comment as some other PRs you've put out with test errors:

https://github.com/flutter/flutter/pull/81629#issuecomment-839991846
> > I am not able to understand that what is exact here
> > can you please guide me?
> 
> Troubleshooting why tests are failing is the responsibility of a PR author, and out of the scope for reviewers.

and https://github.com/flutter/flutter/pull/81629#issuecomment-842594789
> @Abhishek01039 If you're not sure how to debug the test failure you should probably just close this PR. The reviewers aren't likely to prioritize debugging this any time soon. Feel free to re-open with test edits if you figure out why [`web_benchmarks_html`] is failing on this PR.

**Pull request closed without merging**
